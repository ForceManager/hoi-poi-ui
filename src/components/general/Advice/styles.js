export default (theme) => ({
    root: {
        ...theme.typography.defaultText,
        borderRadius: '8px',
        display: 'flex',
        padding: '16px',
        '& .ReactCollapse--collapse': {
            width: '100%',
            minHeight: 20,
            overflow: 'hidden !important',
            transition: 'height 500ms',
            willChange: 'height',
        },
    },
    icon: {
        display: 'flex',
        marginRight: 10,
        marginTop: 1,
    },
    Text: {
        display: 'block',
        width: '100%',
    },

    // Types
    default: {
        backgroundColor: theme.colors.neutral100,
        '& $Text': {
            color: theme.colors.neutral700,
        },
    },
    error: {
        backgroundColor: theme.colors.red100,
        '& $Text': {
            color: theme.colors.red500,
        },
    },
    info: {
        backgroundColor: theme.colors.aqua100,
        '& $Text': {
            color: theme.colors.aqua700,
        },
    },
    success: {
        backgroundColor: theme.colors.green100,
        '& $Text': {
            color: theme.colors.green700,
        },
    },
    warning: {
        backgroundColor: theme.colors.yellow100,
        '& $Text': {
            color: theme.colors.yellow700,
        },
    },
    textContainer: {
        flex: 1,
        overflow: 'hidden',
    },
    collapseContainer: {
        display: 'flex',
        overflow: 'hidden',
        alignItems: 'flex-start',
    },
    dropdownIcon: {
        lineHeight: '0',
        transition: 'transform 0.3s ease-out',
        willChange: 'transform',
        transform: 'rotate(-180deg) translateY(1px)',
        cursor: 'pointer',
    },
    isOpened: {
        '& $dropdownIcon': {
            transform: 'rotate(0) translateY(1px)',
        },
        '& $Text': {
            whiteSpace: 'pre-wrap',
        },
    },
    withTitleContainer: {
        marginTop: 15,
    },
});
