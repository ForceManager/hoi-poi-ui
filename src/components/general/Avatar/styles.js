export default (theme) => ({
    root: {
        ...theme.typography.defaultText,
        color: theme.colors.neutral700,
        overflow: 'hidden',
        background: theme.colors.neutral400,
        position: 'relative',
        '& > img': {
            height: '100%',
            width: '100%',
            objectFit: 'cover',
            textAlign: 'center',
        },
    },
    round: {
        borderRadius: '50%',
        '& $icon': {
            width: '80%',
            height: '80%',
            padding: '10%',
        },
    },
    square: {
        borderRadius: 4,
        '& $icon': {
            width: '96%',
            height: '96%',
            padding: '2%',
        },
    },
    tiny: {
        height: 18,
        width: 18,
        '& $initials': {
            fontSize: 10,
        },
        '& $count': {
            transform: 'translate3d(40%, 40%, 0)',
        },
    },
    small: {
        height: 20,
        width: 20,
        '& $initials': {
            fontSize: 10,
        },
        '& $count': {
            transform: 'translate3d(40%, 40%, 0)',
        },
    },
    medium: {
        height: 25,
        width: 25,
    },
    large: {
        height: 30,
        width: 30,
        '& $initials': {
            fontSize: 16,
        },
    },
    big: {
        height: 32,
        width: 32,
        '& $initials': {
            fontSize: 18,
        },
    },
    huge: {
        height: 64,
        width: 64,
        '& $initials': {
            fontSize: 38,
        },
        '& $count': {
            transform: 'none',
        },
    },
    icon: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    initials: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
        height: '100%',
        textTransform: 'uppercase',
    },
    multiAvatar: {
        overflow: 'visible',
    },
    multiAvatarInner: {
        display: 'grid',
        overflow: 'hidden',
        borderRadius: '50%',
    },
    multiAvatarItem: {
        '& img': {
            width: '100%',
            height: '100%',
            objectFit: 'cover',
            display: 'block',
        },
        '&:first-child': {
            gridColumn: 1,
            gridRow: '1 / span 2',
            aspectRatio: 0.5,
        },

        '&:nth-child(2)': {
            gridColumn: 2,
            gridRow: 1,
            aspectRatio: 1,
        },

        '&:nth-child(3)': {
            gridColumn: 2,
            gridRow: 2,
            aspectRatio: 1,
        },

        '&:nth-child(2):is(:last-child)': {
            gridColumn: 2,
            gridRow: '1 / span 2',
            aspectRatio: 0.5,
        },

        '&:only-child': {
            gridColumn: '1 / span 2',
            gridRow: '1 / span 2',
            aspectRatio: 1,
        },
    },
    count: {
        position: 'absolute',
        bottom: 0,
        right: 0,
        boxSizing: 'border-box',
        boxShadow: `0 0 0 2px ${theme.colors.neutralBase}`,
        backgroundColor: theme.colors.neutral400,
        transform: 'translate3d(25%, 25%, 0)',
    },
});
