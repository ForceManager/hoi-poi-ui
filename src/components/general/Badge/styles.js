export default (theme) => ({
    root: {
        ...theme.typography.smallText,
        lineHeight: '20px',
        color: theme.colors.text.black,
        borderRadius: '3px',
        display: 'inline-block',
        padding: '0 10px',
        minWidth: 70,
        textAlign: 'center',
    },
    Text: {
        lineHeight: '20px',
        fontWeight: 500,
        textTransform: 'uppercase',
    },
    // Types
    error: {
        backgroundColor: theme.colors.redLight,
        '& $Text': {
            color: theme.colors.text.red,
        },
    },
    info: {
        backgroundColor: theme.colors.greyLight,
        '& $Text': {
            color: theme.colors.text.greySoft,
        },
    },
    success: {
        backgroundColor: theme.colors.greenLight,
        '& $Text': {
            color: theme.colors.text.green,
        },
    },
    warn: {
        backgroundColor: theme.colors.yellowLight,
        '& $Text': {
            color: theme.colors.text.yellow,
        },
    },
});
