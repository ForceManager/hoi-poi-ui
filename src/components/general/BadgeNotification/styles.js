export default (theme) => ({
    root: {
        display: 'inline-block',
        textAlign: 'center',
        backgroundColor: theme.colors.orange500,
        position: 'relative',
    },
    Text: {
        padding: '0px 8px',
        verticalAlign: 'middle',
        color: theme.colors.neutralBase,
    },
    medium: {
        height: 24,
        minWidth: 24,
        borderRadius: 24,
        '& $Text': {
            lineHeight: '24px',
        },
    },
    small: {
        height: 18,
        minWidth: 18,
        borderRadius: 18,
        '& $Text': {
            lineHeight: '18px',
        },
    },
});
