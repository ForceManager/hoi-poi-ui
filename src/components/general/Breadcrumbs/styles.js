export default (theme) => ({
    root: {
        display: 'flex',
        alignItems: 'center',
    },
    item: {
        marginRight: 8,
        '& span': {
            color: theme.colors.textLight.secondary,
        },
        '&:last-child': {
            margin: 0,
            '& span': {
                color: theme.colors.textLight.primary,
            },
        },
    },
    clickables: {
        '& $item': {
            cursor: 'pointer',
            '&:hover': {
                '& span': {
                    color: theme.colors.blue500,
                },
                '&:last-child': {
                    '& span': {
                        color: theme.colors.textLight.primary,
                    },
                },
            },
            '&:last-child': {
                cursor: 'inherit',
            },
        },
        '& $divider': {
            cursor: 'inherit',
        },
    },
    divider: {},
});
