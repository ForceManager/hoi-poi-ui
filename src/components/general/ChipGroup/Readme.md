Chips Group:

```jsx
let chipProps = [
    {
        label: 'Lorena Smith',
        url:
            'https://images.pexels.com/photos/2092709/pexels-photo-2092709.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260',
    },
    {
        label: 'Dani Balastegui',
    },
    {
        label: 'Jordi Gandia',
        url:
            'https://images.pexels.com/photos/2169434/pexels-photo-2169434.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
    },
    {
        label: 'John Smith',
        url:
            'https://images.pexels.com/photos/2099225/pexels-photo-2099225.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940',
    },
    {
        label: 'Jordi Gandia',
    },
];

<ChipGroup chips={chipProps} />;
```

### Component tree

---

-   [Chip](#/General?id=chip)
-   root - `Chip` wrapper
-   label - inner content
-   url - string with direct link to the picture
