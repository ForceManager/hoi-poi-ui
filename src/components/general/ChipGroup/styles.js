export default (theme) => ({
    root: {
        '& > *': {
            marginRight: '3px',
        },
    },
});
