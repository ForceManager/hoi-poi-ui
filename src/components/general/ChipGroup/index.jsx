import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'react-jss';
import { getOverrides } from '../../../utils/overrides';

import Chip from './../Chip';
import styles from './styles';

function ChipGroup({
    className,
    overrides: overridesProp,
    classes,
    label,
    url,
    isShrinked,
    chips,
    ...props
}) {
    const override = getOverrides(overridesProp, ChipGroup.overrides);

    return (
        <div className={classes.root} {...override.root}>
            {chips.map((chip, index) => (
                <Chip key={index} url={chip.url} isShrinked={true}>
                    {chip.label}
                </Chip>
            ))}
        </div>
    );
}

Chip.overrides = ['root'];

ChipGroup.defaultProps = {
    className: '',
    overrides: {},
    chips: [],
};

ChipGroup.propTypes = {
    className: PropTypes.string,
    overrides: PropTypes.object,
    chips: PropTypes.arrayOf(
        PropTypes.shape({
            label: PropTypes.string,
            url: PropTypes.string,
        }),
    ),
};

export default React.memo(withStyles(styles, { name: 'ChipGroup', injectTheme: true })(ChipGroup));
