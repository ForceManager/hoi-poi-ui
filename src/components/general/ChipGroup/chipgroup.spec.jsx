import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import ChipGroup from '.';

describe('ChipGroup', () => {
    test('Chips with and without picture', () => {
        const chipProps = [
            {
                label: 'Lorena Smith',
                url: 'https://content-static.upwork.com/uploads/2014/10/01073427/profilephoto1.jpg',
            },
            {
                label: 'Dani Balastegui',
            },
            {
                label: 'Jordi Gandia',
                url:
                    'https://i0.wp.com/zblogged.com/wp-content/uploads/2019/02/FakeDP.jpeg?resize=567%2C580&ssl=1',
            },
            {
                label: 'John Smith',
                url:
                    'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTRp4kKC007NfgaXHB9WubLDpIS5DwEu-qbQkvjY5stXZFYo-dM',
            },
            {
                label: 'Jordi Gandia',
            },
        ];
        const wrapper = mount(
            <HoiPoiProvider>
                <ChipGroup chips={chipProps} />;
            </HoiPoiProvider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
