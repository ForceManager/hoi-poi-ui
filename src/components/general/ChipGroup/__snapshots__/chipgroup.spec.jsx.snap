// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipGroup Chips with and without picture 1`] = `
<Component>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "black": "#262B34",
          "blackLight": "#5f656b",
          "blackberry": "#2e2d4c",
          "blackberryDark": "#24243C",
          "disabledGrey": "#ced4d9",
          "green": "#86AD20",
          "greenLight": "#e7f1d5",
          "grey": "#697782",
          "greyLight": "#f0f3f5",
          "greySoft": "#788590",
          "lines": "#dce3eb",
          "link": "#216db3",
          "linkDark": "#174a79",
          "overlay": "rgba(33,33,33,0.55)",
          "placeholders": "#e1e4e7",
          "placeholdersHover": "#d0d4d8",
          "primary": "#FF8C00",
          "primaryDark": "#C87600",
          "primaryLight": "#fef4e6",
          "primarySoft": "#ff9d26",
          "purple": "#AA5B9C",
          "purpleLight": "#EEDEEB",
          "red": "#f04540",
          "redDark": "#C43833",
          "redLight": "#FAE7E7",
          "redSoft": "#FF544F",
          "secondary": "#00A8Be",
          "secondaryGrey": "#e4eeff",
          "secondaryLight": "#D6EEF2",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "secondarySoft": "#AADFE7",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "tableHeader": "#f6f7f9",
          "tableHeaderHover": "#EBECEF",
          "tableHover": "#FAFBFC",
          "text": Object {
            "black": "#3c4248",
            "danger": "white",
            "default": "#697782",
            "green": "#86AD20",
            "greySoft": "#788590",
            "primary": "white",
            "red": "#f04540",
            "secondary": "#00A8Be",
            "white": "#FFFFFF",
            "yellow": "#E9B200",
          },
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "white": "#FFFFFF",
          "yellow": "#E9B200",
          "yellowDark": "#BE9100",
          "yellowLight": "#FCF7E5",
        },
        "effects": Object {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "buttonActive": Object {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": Object {
          "bigText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "button": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "lineHeight": 1.5,
          },
          "defaultText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 13,
            "letterSpacing": 0.5,
          },
          "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          "fontSizeBase": 13,
          "fontSizeSmall": 11,
          "largeText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "smallText": Object {
            "fontFamily": "\\"Roboto\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": Object {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": Object {
              "minify": false,
            },
            "plugins": Array [],
          },
          "plugins": PluginsRegistry {
            "plugins": Object {
              "external": Array [
                Object {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onProcessRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": Array [
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
                Object {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": Object {
              "onChangeValue": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": Array [],
              "onProcessStyle": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": Array [
                [Function],
              ],
            },
          },
          "version": "10.0.0",
        }
      }
    >
      <ChipGroup
        chips={
          Array [
            Object {
              "label": "Lorena Smith",
              "url": "https://content-static.upwork.com/uploads/2014/10/01073427/profilephoto1.jpg",
            },
            Object {
              "label": "Dani Balastegui",
            },
            Object {
              "label": "Jordi Gandia",
              "url": "https://i0.wp.com/zblogged.com/wp-content/uploads/2019/02/FakeDP.jpeg?resize=567%2C580&ssl=1",
            },
            Object {
              "label": "John Smith",
              "url": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTRp4kKC007NfgaXHB9WubLDpIS5DwEu-qbQkvjY5stXZFYo-dM",
            },
            Object {
              "label": "Jordi Gandia",
            },
          ]
        }
        className=""
        overrides={Object {}}
      >
        <div
          className="HoiPoi__ChipGrou__root-1-0"
        >
          <Chip
            className=""
            isShrinked={true}
            key="0"
            overrides={Object {}}
            url="https://content-static.upwork.com/uploads/2014/10/01073427/profilephoto1.jpg"
          >
            <div
              className="HoiPoi__Chip__root-1-1"
            >
              <div
                className="HoiPoi__Chip__wrapper-1-2"
              >
                <img
                  alt=""
                  className="HoiPoi__Chip__shrinkedPicture-1-7"
                  src="https://content-static.upwork.com/uploads/2014/10/01073427/profilephoto1.jpg"
                />
              </div>
            </div>
          </Chip>
          <Chip
            className=""
            isShrinked={true}
            key="1"
            overrides={Object {}}
            url=""
          >
            <div
              className="HoiPoi__Chip__root-1-1"
            >
              <div
                className="HoiPoi__Chip__wrapper-1-2"
              >
                <Text
                  className="HoiPoi__Chip__shrinked-1-4"
                  isTruncated={false}
                  overrides={Object {}}
                  size="medium"
                  type="normal"
                >
                  <span
                    className="HoiPoi__Text__root-1-8 HoiPoi__Text__normal-1-15 HoiPoi__Text__medium-1-11 HoiPoi__Chip__shrinked-1-4"
                  >
                    D
                    B
                  </span>
                </Text>
              </div>
            </div>
          </Chip>
          <Chip
            className=""
            isShrinked={true}
            key="2"
            overrides={Object {}}
            url="https://i0.wp.com/zblogged.com/wp-content/uploads/2019/02/FakeDP.jpeg?resize=567%2C580&ssl=1"
          >
            <div
              className="HoiPoi__Chip__root-1-1"
            >
              <div
                className="HoiPoi__Chip__wrapper-1-2"
              >
                <img
                  alt=""
                  className="HoiPoi__Chip__shrinkedPicture-1-7"
                  src="https://i0.wp.com/zblogged.com/wp-content/uploads/2019/02/FakeDP.jpeg?resize=567%2C580&ssl=1"
                />
              </div>
            </div>
          </Chip>
          <Chip
            className=""
            isShrinked={true}
            key="3"
            overrides={Object {}}
            url="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTRp4kKC007NfgaXHB9WubLDpIS5DwEu-qbQkvjY5stXZFYo-dM"
          >
            <div
              className="HoiPoi__Chip__root-1-1"
            >
              <div
                className="HoiPoi__Chip__wrapper-1-2"
              >
                <img
                  alt=""
                  className="HoiPoi__Chip__shrinkedPicture-1-7"
                  src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTRp4kKC007NfgaXHB9WubLDpIS5DwEu-qbQkvjY5stXZFYo-dM"
                />
              </div>
            </div>
          </Chip>
          <Chip
            className=""
            isShrinked={true}
            key="4"
            overrides={Object {}}
            url=""
          >
            <div
              className="HoiPoi__Chip__root-1-1"
            >
              <div
                className="HoiPoi__Chip__wrapper-1-2"
              >
                <Text
                  className="HoiPoi__Chip__shrinked-1-4"
                  isTruncated={false}
                  overrides={Object {}}
                  size="medium"
                  type="normal"
                >
                  <span
                    className="HoiPoi__Text__root-1-8 HoiPoi__Text__normal-1-15 HoiPoi__Text__medium-1-11 HoiPoi__Chip__shrinked-1-4"
                  >
                    J
                    G
                  </span>
                </Text>
              </div>
            </div>
          </Chip>
        </div>
      </ChipGroup>
      ;
    </JssProvider>
  </ThemeProvider>
</Component>
`;
